### ISSUE TITLE
quarto embeds captions in figures

### ISSUE THREAD

**Issue by ajrgodfrey** [quarto-dev/quarto-web#253]
I've created figures using R and mermaid. The outcomes for the figures differs and I suspect neither meets WCAG.

Both code blocks use label, fig-cap, and fig-alt.

The R graph does get an alt tag, created using #| fig-alt: ...
It is read aloud by my screen reader when I get to the graphic's placement in the text.
However, the caption is not read aloud by either  screen reader I've tried.Initially, I thought it was because the caption is being embedded int he image file, but upon inspection of the png in the figures subfolder, the image does not include the caption.

WRT the mermaid flow chart: I do not get a caption with the same findings as for the R graph. I do not though even get told the graphic exists in my html page by the screen readers nor is the alt tag text read aloud. I have tried using hte default presentation of the mermaid graph, as well as explicitly choosing to create png and svg; the outcomes remain the same.

I've really appreciated the inclusion of fig.alt in knitr, as an opportunity to get more information than can be imparted in a figure's caption.

How can I get the captions included as regular text as was the norm with rmarkdown::render()?

Is there a switch to add to the main YAML? If so, I haven't found it documented. If not, can I request that one be created.
Thanks,
Jonathan

**Comment by batpigandme** [comment]
Hi @ajrgodfrey,

Thanks for this find. 

From inspecting the HTML for the example page with diagrams ([Quarto Authoring > Diagrams](https://quarto.org/docs/authoring/diagrams.html)), it looks like the caption is getting hidden by `aria-hidden="true"`. (Apologies for not linking to the HTML source file—the output directory for the site is in `.gitignore`, so the HTML isn't on here).

```html
<figcaption aria-hidden="true" class="figure-caption">Figure 1: This is a simple graphviz graph</figcaption>
```

I _think_ this is the default behavior for captions in Quarto—the assumption being that to have both the alt text _and_ the caption would be duplicative (again, I'm kind of guessing here), and it sounds like this isn't always the desired behavior.

While we should definitely correct this on the docs site, I think it's worth addressing this in the source code that generates this (i.e. in the [quarto-cli repo](https://github.com/quarto-dev/quarto-cli)).  Since these are part of the same organization, it should be possible to migrate this issue there for someone with the proper permissions (cc @jjallaire, @dragonstyle).

Thanks again for filing and finding this issue.

(pinging @jooyoungseo, too, since this'll definitely go with the [accessibility issues](https://github.com/quarto-dev/quarto-cli/issues?q=is%3Aissue+is%3Aopen+label%3Aaccessibility+))

**Comment by jjallaire** [comment]
It looks like we are doing this b/c we observed the `aria-hidden="true"` in HTML generated by pandoc and were trying to match it (note however that our behavior is faulty b/c pandoc does this only when it includes alt text that matches the caption):

```html
<figure>
<img src="image.png" alt="Woozy" />
<figcaption aria-hidden="true">Woozy</figcaption>
</figure>
```

In our case we are outputting this unqualified whenever we generate an HTML figcaption tag. The most straightforward thing we can do immediately is just stop outputting the area-hidden attribute in our own HTML (that is a safe enough change for us to make late in v1.0 development). Something fancier where we have more dynamic behavior and/or add attributes to allow the user to control the behavior we can't do right now but will certainly entertain after v1.0.

So the simple question is: should I remove the code that generates `aria-hidden="true"` ?

**Comment by jjallaire** [comment]
I also noticed that we are worse than default pandoc b/c we will add the `aria-caption="hidden"` _without_ including the `<figcaption>` text as the `alt` attribute on the image:

```html
<figure class="figure">
<p><img src="image.png" class="img-fluid figure-img"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Woozy</figcaption><p></p>
</figure>
```

This has me pretty convinced that we should just remove the code that generates `aria-hidden="true"` for now and then revisit with more intelligent / user-configurable behavior post v1.0.
